.github/actions/image-upload/action.yml

```diff
@@ -45,20 +45,19 @@ runs:
         IMG_DIR=${{ inputs.image_dir }}
         VERSION=$(cat "${SRC_DIR}/serial")
 
-        # Create directory structure that will be mirrored on the target server.
-        PRODUCT_PATH="${SRC_DIR}-upload/${IMG_DIR}"
-        mkdir -p "${PRODUCT_PATH}/${VERSION}"
-
-        # Move config.yaml file if exists.
-        if [ -f "${SRC_DIR}/config.yaml" ]; then
-            mv "${SRC_DIR}/config.yaml" "${PRODUCT_PATH}"
-        fi
+        # First upload contents to the temporary dir and once fully uploaded
+        # move the directory to the final destination to avoid potential race
+        # where simplestream-maintainer includes partially uploaded images.
+        TMP_DIR="${IMG_DIR}/.${VERSION}"
+        DST_DIR="${IMG_DIR}/${VERSION}"
 
-        # Move image content.
-        mv ${SRC_DIR}/* "${PRODUCT_PATH}/${VERSION}"
+        # Create directory structure that will be mirrored on the target server.
+        mkdir -p "${SRC_DIR}-upload/${TMP_DIR}"
+        mv ${SRC_DIR}/* "${SRC_DIR}-upload/${TMP_DIR}"
 
         # Use SFTP to upload images to the server.
         sftp -P ${SSH_PORT} "${SSH_USER}@${SSH_HOST}" <<EOF
             put -r ${SRC_DIR}-upload/*
```

Comment:
`man sftp` doesn't mention `put -r`, only `put -R` so while both apparently work, I'd switch to the officially documented lower case one. I know `scp` uses `-r`, confusing :/

Reply:
Yep, `put -R` seems to be what's in the man page.